nir/src/main/scala/scala/scalanative/nir/Global.scala

```diff
@@ -67,4 +67,16 @@ object Global {
 
   def toProxySignature(signature: String) = signature + "_proxy"
 
+  def stripImplClassTrailingDollar(name: Global): Global = name match {
+    case Global.None =>
+      name
+    case Global.Top(id) =>
+      if (id.endsWith("$class$")) {
+        Global.Top(id.substring(0, id.length - 1))
+      } else {
+        name
+      }
+    case Global.Member(subname, id) =>
+      Global.Member(stripImplClassTrailingDollar(subname), id)
+  }
```

Comment:
@sjrd So a potential workaround: if a symbol name ends with "$class$", automatically strip the last dollar. The file name still contains a dollar but the tree that's deserialized from it is not going to have it. I tried it with library from 0.3.6 and a 0.3.7-compiled application and it seems to work.

Reply:
Seems plausible.